Svc/Deframer/Deframer.cpp

```diff
@@ -0,0 +1,306 @@
+    // Check that this is the first time we are calling setup
+    FW_ASSERT(m_protocol == nullptr);
+    // Assign the protocol passed in to m_protocol
+    m_protocol = &protocol;
+    // Pass *this as the DeframingProtocolInstance to protocol setup
+    // Deframer is derived from and implements DeframingProtocolInterface
+    protocol.setup(*this);
+}
+
+// ----------------------------------------------------------------------
+// Handler implementations for user-defined typed input ports
+// ----------------------------------------------------------------------
+
+void Deframer ::cmdResponseIn_handler(
+    NATIVE_INT_TYPE portNum,
+    FwOpcodeType opcode,
+    U32 cmdSeq,
+    const Fw::CmdResponse& response
+) {
+    // Nothing to do
+}
+
+void Deframer ::framedIn_handler(
+    const NATIVE_INT_TYPE portNum,
+    Fw::Buffer& recvBuffer,
+    const Drv::RecvStatus& recvStatus
+) {
+    // Check whether there is data to process
+    if (Drv::RecvStatus::RECV_OK == recvStatus.e) {
+        // There is: process the data
+        processBuffer(recvBuffer);
+    }
+    // Deallocate the buffer
+    framedDeallocate_out(0, recvBuffer);
+}
+
+void Deframer ::schedIn_handler(
+    const NATIVE_INT_TYPE portNum,
+    NATIVE_UINT_TYPE context
+) {
+    // Check for data
+    Fw::Buffer buffer(m_pollBuffer, sizeof(m_pollBuffer));
+    Drv::PollStatus status = framedPoll_out(0, buffer);
+    if (status == Drv::PollStatus::POLL_OK) {
+        // Data exists: process it
+        processBuffer(buffer);
+    }
+}
+
+// ----------------------------------------------------------------------
+// Implementation of DeframingProtocolInterface
+// ----------------------------------------------------------------------
+
+Fw::Buffer Deframer ::allocate(const U32 size)  {
+    return bufferAllocate_out(0, size);
+}
+
+void Deframer ::route(Fw::Buffer& packetBuffer) {
+
+    // Read the packet type from the packet buffer
+    FwPacketDescriptorType packetType = Fw::ComPacket::FW_PACKET_UNKNOWN;
+    Fw::SerializeStatus status = Fw::FW_SERIALIZE_OK;
+    {
+        Fw::SerializeBufferBase& serial = packetBuffer.getSerializeRepr();
+        status = serial.setBuffLen(packetBuffer.getSize());
+        FW_ASSERT(status == Fw::FW_SERIALIZE_OK);
+        status = serial.deserialize(packetType);
+    }
+
+    // Whether to deallocate the packet buffer
+    bool deallocate = true;
+
+    // Process the packet
+    if (status == Fw::FW_SERIALIZE_OK) {
+        U8 *const packetData = packetBuffer.getData();
+        const U32 packetSize = packetBuffer.getSize();
+        switch (packetType) {
+            // Handle a command packet
+            case Fw::ComPacket::FW_PACKET_COMMAND: {
+                // Allocate a com buffer on the stack
+                Fw::ComBuffer com;
+                // Copy the contents of the packet buffer into the com buffer
+                status = com.setBuff(packetData, packetSize);
+                if (status == Fw::FW_SERIALIZE_OK) {
```

Comment:
We don't check that it is connected here?  Is the deframer required to have a command connection? We specifically check the connection on the buffer port.

That is: this component can be used w/o buffer connections, but cannot be used w/o comm buffer connections.  Is this asymmetry ok?

Reply:
This is how the inherited code was, and the behavior is captured in the requirements. 

https://github.com/bocchino/fprime/blob/deframer-design-code/Svc/Deframer/docs/sdd.md#3-requirements

See SVC-DEFRAMER-011.